### PR TITLE
Exclude feature

### PR DESCRIPTION
**Pull request by cyrinux**
Add --exclude=pattern1,pattern2 params.

Pattern are not regex just word to exclude from the menu

### CODE REVIEW

File: src/Main.hh [hunk 1]
```diff
@@ -139,11 +172,12 @@ private:
                 {"help",    no_argument,        0,  'h'},
                 {"display-binary", no_argument, 0,  'b'},
                 {"no-generic", no_argument,     0,  'n'},
+                {"exclude", required_argument,  0,  'q'},
```
Comment (enkore) [inline src/Main.hh]: `'e'` would be more natural I guess

File: src/Main.hh [hunk 2]
```diff
@@ -253,10 +290,14 @@ private:
...
     std::string dmenu_command;
     std::string terminal;
+    std::string exclude;
 
     stringlist_t environment;
-    bool use_xdg_de = false;
+    stringlist_t patterns;
+
     bool exclude_generic = false;
+    bool to_exclude = false;
```
Comment (enkore) [inline src/Main.hh]: Since this only seems to be used as a local variable, make it one, too?

File: src/Main.hh [hunk 3]
```diff
@@ -73,12 +73,43 @@ public:
...
-            const std::string &generic_name = app.second->generic_name;
-            if(!exclude_generic && !generic_name.empty() && app.second->name != generic_name)
-                this->dmenu->write(generic_name);
+	    const std::string app_name = app.second->name;
+	    const std::string app_gen_name = app.second->generic_name;
+
+	    std::string lower_app_name = app_name;
+	    std::string lower_app_gen_name = app_gen_name;
+	    std::transform(lower_app_name.begin(), lower_app_name.end(), lower_app_name.begin(), ::tolower);
+	    std::transform(lower_app_gen_name.begin(), lower_app_gen_name.end(), lower_app_gen_name.begin(), ::tolower);
```
Comment (enkore) [inline src/Main.hh]: Since this block does some work, put it (and the following loop) into an `if` that only runs if exclude patterns were specified?

**Comment by cyrinux** [comment]
Thanks you, I'm a beginner thanks for your report I will try to improve
this with your comment :)

**Comment by cyrinux** [comment]
What do think about my new commit?

**Pull request closed without merging**